lms/services/application_instance.py

```diff
@@ -116,6 +116,19 @@ def search(
 
         return query.all()
 
+    def set_canvas_api_details(
+        self, application_instance, developer_key, developer_secret
+    ):
```

Comment:
Although this happens to only deal with Canvas stuff, is this actually the start of an `update_application_instance()`?

Reply:
Done this but felt like I should include other fields we do upgrade instead of using two different method to update AI fields.

Reply:
One way to express the difference between being passed None and not being passed something is to use something like `...`. This is pretty much good enough:

```python
def update_application_instance(self, lms_url=...):
    if lms_url is not ...:
        # Do whatever
```

In the out-there world where you might actually want to store `...` you can create an alternative singleton like `MISSING = object()` and then use that instead.